doc/tests.rst

```diff
@@ -448,8 +449,160 @@ which displays nicer looking diffs that the standard ``diff``
 tool. You can change that by passing ``--diff-command CMD`` to
 dune.
 
+Cram Tests
+==========
+
+Cram tests are expectation tests written in a shell-like syntax. They are ideal
+for testing binaries. Cram tests are auto discovered from files or directories
+with a ``.t`` extension, so they must be enabled manually in the
+``dune-project`` file:
+
+.. code:: scheme
+
+   (lang dune 2.7)
+   (cram enable)
+
+
+File Tests
+----------
+
+To define a standalone test, we create a ``.t`` file. For example, ``foo.t``:
+
+.. code:: bash
+   Simplest possible cram test
+     $ echo "testing"
+
+This simple example demonstrates two components of cram tests:
+
+* Comments - Anything that doesn't start with a 2 space indentation is a comment
+
+* Commands - A command starts with 2 spaces followed by a ``$``. It is executed
+  in the shell and the output is diffed against the output below. In this
+  example, there's no output yet.
+
+To run the test and promote the results:
+
+.. code:: bash
+
+   $ dune runtest
+   $ dune promote
+
+We now see the output of the command:
+
+.. code:: bash
+
+   Simplest possible cram test
+     $ echo "testing"
+     testing
+
+This is the main advantage of expect tests. We don't need to write assertions
+manually, instead we detect failure when the command produces a different output
+than what is recorded in the test script.
+
+For example, here's an example of how we'd test the ``wc`` utility. ``wc.t``:
+.. code:: bash
+
+   We create a fixture called foo
+     $ cat >foo <<EOF
+     > foo
+     > bar
+     > baz
+     > EOF
+
+   After creating the fixture, we want to verify that ``wc`` gives us the right
+   result:
+     $ wc -l foo | awk '{ print $1 }'
+     4
+
+The above example uses the here doc syntax to pipe the subsequent lines to
+``cat``. This is convenient for creating small fixtures.
```

Comment:
I didn't know what fixture meant. After googling it, it seems like the right word. If it was just me, we can leave it, but if it is a word that is not very common, I suggest to stick to simpler language to keep the documentation accessible. Here we could use "auxiliary file" for instance.

Reply:
It's a common name in the python community.

I can just refer the term with "test artifacts" instead. I think that's a bit more self explanatory.